src/github_runner_manager/reactive/consumer.py

```diff
@@ -92,37 +117,46 @@ def consume(
 
     Raises:
         JobError: If the job details are invalid.
+        QueueError: If an error when communicating with the queue occurs.
     """
-    with Connection(queue_config.mongodb_uri) as conn:
-        with closing(SimpleQueue(conn, queue_config.queue_name)) as simple_queue:
-            with signal_handler(signal.SIGTERM):
-                msg = simple_queue.get(block=True)
-                try:
-                    job_details = cast(JobDetails, JobDetails.parse_raw(msg.payload))
-                except ValidationError as exc:
-                    msg.reject(requeue=True)
-                    raise JobError(f"Invalid job details: {msg.payload}") from exc
-                logger.info(
-                    "Received job with labels %s and job_url %s",
-                    job_details.labels,
-                    job_details.url,
-                )
-                if not _validate_labels(
-                    labels=job_details.labels, supported_labels=supported_labels
-                ):
-                    logger.error(
-                        "Found unsupported job labels in %s. "
-                        "Will not spawn a runner and requeue the message.",
+    try:
+        with Connection(queue_config.mongodb_uri) as conn:
+            with closing(SimpleQueue(conn, queue_config.queue_name)) as simple_queue:
+                with signal_handler(signal.SIGTERM):
+                    msg = simple_queue.get(block=True)
+                    try:
+                        job_details = cast(JobDetails, JobDetails.parse_raw(msg.payload))
+                    except ValidationError as exc:
+                        logger.error("Found invalid job details, will reject the message.")
+                        msg.reject(requeue=False)
+                        raise JobError(f"Invalid job details: {msg.payload}") from exc
```

Comment:
I'm wondering if this should just log an error and leave it to the users to debug it via grafana (later on)?

Reply:
@yanksyoon Do you mean just logging an error rather than throwing an exception? Can you explain the reasoning please.

Reply:
I was wondering if the exception would put the charm in errored state and if any job misconfiguration is expected to put the charm in errored state (since it will go in a error retry loop)?

Reply:
Oh, it won't be put in error state and retried, because it is run in a separate process in the background. The script will just error out and terminate.

Reply:
Am resolving the conversation now, feel free to ping me or re-open.